src/common/constants.ts

```diff
@@ -25,10 +27,14 @@ export const domains = [
     'deriv-com-git-fork',
 ] //deriv-com-git-fork for vercel server, localhost - for developer mode
 
-const getDomainUrl = (): string =>
-    isBrowser() && window.location.hostname in domain_url_pair
-        ? domain_url_pair[window.location.hostname]
+const getDomainUrl = (): string => {
+    const var_staging_flag = isBrowser() ? localStorage.getItem('staging_demo') : null
+    const var_staging =
+        isBrowser() && null === var_staging_flag ? window.location.hostname : 'staging.deriv.com'
+    return isBrowser() && var_staging in domain_url_pair
+        ? domain_url_pair[var_staging]
         : deriv_com_url
+}
```

Comment:
```suggestion
const getDomainUrl = (): string => {
    const staging_flag = localStorage.getItem('staging_demo') || null
    const current_url = staging_flag ? 'staging.deriv.com' : window.location.hostname 
    
    return isBrowser() && current_url in domain_url_pair
        ? domain_url_pair[current_url]
        : deriv_com_url
}
```


let's keep the variable names meaningful and simplier